### PR TITLE
Updated to BEST+PUPPI

### PR DESCRIPTION
**Pull request by justinrpilot**
I updated the code to include the new 6-category BEST NN which is trained on PUPPI jets.  Several path names changed to point to the new JSON file in data/BEST/.  Also a few changes to the BoostedEventShapeTagger.cc to fix how the jet constituents are used.

### CODE REVIEW

File: src/BoostedEventShapeTagger.cc [hunk 1]
```diff
@@ -243,6 +245,18 @@ void BoostedEventShapeTagger::getJetValues( const pat::Jet& jet ){
...
 
+
+	
+
+	thisParticleLV_jet *= puppiWt;
+        thisParticleLV_top *= puppiWt;
+        thisParticleLV_W *= puppiWt;
+        thisParticleLV_Z *= puppiWt;
+        thisParticleLV_H *= puppiWt;
+        if (daughter->pt() > 1.0)
```
Comment (hqucms) [inline src/BoostedEventShapeTagger.cc:256]: So here it is the original pt, not `puppiWt*daughter->pt()`?

File: src/BoostedEventShapeTagger.cc [hunk 2]
```diff
@@ -482,7 +496,7 @@ void BoostedEventShapeTagger::getJetValues( const pat::Jet& jet ){
     m_BESTvars["et"]      = thisJet.Pt();
     m_BESTvars["eta"]     = thisJet.Rapidity();
     m_BESTvars["mass"]    = thisJet.M();
-    m_BESTvars["SDmass"]  = jet.userFloat("ak8PFJetsPuppiSoftDropMass");
+    m_BESTvars["SDmass"]  = jet.mass();
```
Comment (hqucms) [inline src/BoostedEventShapeTagger.cc]: So here you don't use the softdrop mass anymore? Which one did you from MiniAOD?

File: src/BoostedEventShapeTagger.cc (hunk 1 above)
Comment (justinrpilot) [reply src/BoostedEventShapeTagger.cc:256]: Yes, it was trained on the charge without the puppiWt factor.  (Could be updated in the future but it is consistent with the training as-is.

Comment (hqucms) [reply src/BoostedEventShapeTagger.cc:256]: OK, great!

File: src/BoostedEventShapeTagger.cc (hunk 2 above)
Comment (hqucms) [reply src/BoostedEventShapeTagger.cc]: Or do you use "ak8PFJetsPuppiValueMap:mass" from MiniAOD? Then I think it is the same as `jet.mass()` here.

Comment (justinrpilot) [reply src/BoostedEventShapeTagger.cc]: I was testing different masses and may have switched to   ````ak8PFJetsPuppiValueMap:mass ```` by accident.  The training was done with:  
````float puppiSDmass = (puppi_subjet0+puppi_subjet1).M();````

Comment (justinrpilot) [reply src/BoostedEventShapeTagger.cc]: Sorry for the confusion, in the midst of checking the inputs I thought there was some difference in the mass, and was testing different userFloats.  I did another check just now and get consistent results with the soft drop mass (it was consistent before as I changed the inputs in both places to the raw PUPPI mass).  I reverted this change.

Comment (hqucms) [reply src/BoostedEventShapeTagger.cc]: @justinrpilot 

I think if you use `float puppiSDmass = (puppi_subjet0+puppi_subjet1).M();` for the training, then the most consistent one would be `jet.groomedMass()` [0] which are also computed from the subjets (`ak8PFJetsPuppiSoftDropMass` is computed from subjets before applying the JEC though).

[0] https://github.com/cms-sw/cmssw/blob/master/DataFormats/PatCandidates/interface/Jet.h#L527

Comment (hqucms) [reply src/BoostedEventShapeTagger.cc]: I realized that the puppi subjets stored in MiniAOD does not have the latest JEC applied as the subjets reclustered by JetToolBox, so if you have already seen good agreement with `ak8PFJetsPuppiSoftDropMass` then maybe it's better to just stick to that. I expect the effect on the final DNN output is anyhow very small, right?

Comment (justinrpilot) [reply src/BoostedEventShapeTagger.cc]: Yes, I saw good agreement with ````ak8PFJetsPuppiSoftDropMass```` -- it should have a small effect and I agree we can go ahead with this.

**Pull request closed without merging**